packages/db/src/search/AsyncOperationSearch.ts

```diff
@@ -0,0 +1,82 @@
+import { Knex } from 'knex';
+import pick from 'lodash/pick';
+
+import { ApiAsyncOperation } from '@cumulus/types/api/async_operations';
+import Logger from '@cumulus/logger';
+
+import { BaseSearch } from './BaseSearch';
+import { DbQueryParameters, QueryEvent } from '../types/search';
+import { PostgresAsyncOperationRecord } from '../types/async_operation';
+import { translatePostgresAsyncOperationToApiAsyncOperation } from '../translate/async_operations';
+
+const log = new Logger({ sender: '@cumulus/db/AsyncOperationSearch' });
+
+/**
+ * Class to build and execute db search query for asyncOperation
+ */
+export class AsyncOperationSearch extends BaseSearch {
+  constructor(event: QueryEvent) {
+    super(event, 'asyncOperation');
+  }
+
+  /**
+   * Build basic query
+   *
+   * @param knex - DB client
+   * @returns queries for getting count and search result
+   */
+  protected buildBasicQuery(knex: Knex)
+    : {
+      countQuery: Knex.QueryBuilder,
+      searchQuery: Knex.QueryBuilder,
+    } {
+    const countQuery = knex(this.tableName)
+      .count('*');
+
+    const searchQuery = knex(this.tableName)
+      .select(`${this.tableName}.*`);
+    return { countQuery, searchQuery };
+  }
+
+  /**
+   * Build queries for infix and prefix
+   *
+   * @param params
+   * @param params.countQuery - query builder for getting count
+   * @param params.searchQuery - query builder for search
+   * @param [params.dbQueryParameters] - db query parameters
+   */
+  protected buildInfixPrefixQuery(params: {
+    countQuery: Knex.QueryBuilder,
+    searchQuery: Knex.QueryBuilder,
+    dbQueryParameters?: DbQueryParameters,
+  }) {
+    const { countQuery, searchQuery, dbQueryParameters } = params;
+    const { infix, prefix } = dbQueryParameters ?? this.dbQueryParameters;
+    if (infix) {
+      [countQuery, searchQuery].forEach((query) => query.whereRaw(`${this.tableName}.id::text like '%${infix}%'`));
+    }
+    if (prefix) {
+      [countQuery, searchQuery].forEach((query) => query.whereRaw(`${this.tableName}.id::text like '${prefix}%'`));
+    }
+  }
+
+  /**
+   * Translate postgres records to api records
+   *
+   * @param pgRecords - postgres records returned from query
+   * @returns translated api records
+   */
+  protected translatePostgresRecordsToApiRecords(pgRecords: PostgresAsyncOperationRecord[])
```

Comment:
Is there a reason this isn't a non-class method in /src/translate/* or src/lib?  This feels less like a search method and more like a translation helper that isn't specific to this class.

Reply:
Nit/Suggestion: 

-) make fields `const { fields } = this.dbQueryParameters;` instead of replicating `this.dbQueryParameters.fields`  (multi-line ternaries)

Reply:
[55506db](https://github.com/nasa/cumulus/commit/55506db4ba72135467f07c10d16fb3b77797e9bb)

Reply:
translatePostgresRecordsToApiRecords in some case not only does translation but also performs additional query https://github.com/nasa/cumulus/blob/feature/es-phase-2/packages/db/src/search/CollectionSearch.ts#L186

Reply:
Hmm.... fair enough.   I was wondering if we shouldn't abstract this further, as the function does two things, but that's a relatively minor concern here.